server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -101,7 +102,9 @@
     public static final Setting<Long> INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING =
         Setting.longSetting("index.mapping.total_fields.limit", 1000L, 0, Property.Dynamic, Property.IndexScope);
     public static final Setting<Long> INDEX_MAPPING_DEPTH_LIMIT_SETTING =
-            Setting.longSetting("index.mapping.depth.limit", 20L, 1, Property.Dynamic, Property.IndexScope);
+        Setting.longSetting("index.mapping.depth.limit", 20L, 1, Property.Dynamic, Property.IndexScope);
+    public static final Setting<Long> INDEX_MAPPING_FIELD_NAME_LENGTH_LIMIT_SETTING =
+        Setting.longSetting("index.mapping.field.name.length.limit", Long.MAX_VALUE, 15L, Property.Dynamic, Property.IndexScope);
```

Comment:
While I think a minimum length of 15 makes much sense, I would probably set the allowed minimum to 1. Some of the other minimum values here also are not very practical, but by setting it to something larger this leads to question like "why not 10, 5, 3?".

Reply:
Hi Christoph, Thanks for reviewing.  I forgot to mention this when opening the PR but the current check is quite blunt. If a user sets the limit to a low value,
such as 1, a number of templates (from watcher and monitoring plugins) that are set up automatically on server startup will fail. I can look into finding a method to 
skip the length check for system created mappers if you think that is needed? Also it checks `MetaDataFieldMapper`s field name length, I could put an `instanceof` check in to skip these also if needed.

Reply:
This sounds strange. I was under the impression this setting (if set at all), only applies to a certain index. Are you saying that system-level indices and templates (like .watcher etc...) would be effected by this even though if a user changes this for index `fooBar` only?

Reply:
Thanks that makes sense, I was testing it incorrectly.